python/py-hgsubversion/Portfile

```diff
@@ -17,9 +17,9 @@ description             hgsubversion is a Mercurial extension for \
                         working with Subversion (svn) repositories.
 long_description        ${description}
 
-checksums               rmd160  63318040765bbf016275dc98d2b65db70d03b897 \
-                        sha256  da3b4d0450579233bcec55c3b3c365e260d4b48bb53ad35ccc3ab3d0200a3383 \
-                        size    304824
+checksums               rmd160  08f1be76912ae81a56c8eb49e96d12bd47916742 \
+                        sha256  daec4abab484ec63956964e18af36f401ba5f5b12981ffd2858b8466aac5e7ab \
+                        size    306205
 
 # can't set python.versions before adding custom subports
 subport py-hgsubversion-devel {}
```

Comment:
You probably want to delete (`replaced_by`) that one as well? If @seanfarley abandoned mercurial, I assume that his maintainership of `py27-hgsubversion-devel` is not exactly up-to-date either?

Reply:
probably also the other *-devel, like py-hgevolve-devel. I think that if nomaintainer is set, we should not have -devel. Anyway ```py27-hgsubversion-devel``` is old.

Reply:
Yes. Unless there is a good reason, I would say that all these devel ports should go.

The mainatiner resigned a week ago, but even before that the devel ports were older as he didn't have sufficient time to look over them. In case you would like to become a new maintainer, just add your name.

Reply:
I *may* volunteer to maintain the `hg-hggit-devel` port. The reason being that new releases of Mercurial frequently break the `hg-git` extension, and many times there's a fix soon after committed to the latter's `master`.

Because I use this extension on a daily basis, I could make a best effort to bump `-devel` when I notice breakage. But on the other hand, what use is `py-hggit` if it does not work for a long time when `mercurial` is updated? Sometimes it takes weeks/months for the extension to have a new official release. Maybe we should simply not have `-devel` and either bump or backport the fix in these scenarios.

Reply:
We use devel versions when we want to use some experimental feature or test whether the latest version works before the release, so that potential bugs can be fixed before the release.

If the main port is broken, we need to fix the main port, not use the devel port as a replacement. But feel free to volunteer for `py-hggit`.

Reply:
I would argue that it is not as clear-cut in this case, since in the scenario I explained the latest (hg-git) upstream release *is* broken, and only the unreleased development version is fixed. In the past I've seen cases where the fix in `master` is not released for a quarter or so and the latest official release remains broken during this whole period. (This is what motivated my previous bumps to the `py-hggit-devel`.)

Should a port reflect the latest official release, even if it's broken, or an upstream development snapshot? My understanding is that normally ports should point to upstream releases, and `-devel` to unreleased snapshots. If we want to prioritize this consistency above everything else then in this scenario `-devel` is the one that should contain the fix. (I'm not necessarily arguing that this is what we should do.)

P.S.: This discussion about `hg-git` might be off-topic for this PR, if you agree I would gladly take it elsewhere.

Reply:
ok. now I understand why of ```-devel```.  For the moment I updated ```py-hggit-devel``` because fixes some issues with mercurial 4.8.
Use ```master``` as main track it is not a good idea, a least for me. My idea is: remove all ```-devel``` from unmaintained  ports and leave it only on ```py-hggit``` where, after this PR, you will take the ownership.

Reply:
I still argue that if upstream release is broken, we don't do any favour to our users to keep a broken port, even if we provide a working `-devel` alternative. That means extra non-intuitive work for users. I would either use master (say at the first commit that makes the software usable), or backport the changes. The first thing is usually easier. We usually try to stick to releases because:
* we barely keep up with official releases, forget about upgrading to every single commit of every single software we ship, and then additionally test that
* upstream knows best the judgement about when code is ready for public use

But if the release is broken, no release happens for months or years, and master works ... we switch to master, that's no-brainer, I've done that to plenty other ports. I pick a version that's known to be lower than the next expected release, even if that means a bit of cheating, I don't care. Alternative is not to update mercurial, but I guess that's relatively bad.

Reply:
I agree with you, @mojca. I just wanted to describe the potential violation of port naming customs, but I do agree that for `py-hggit` we should open an exception when upstream gets broken.

I didn't want to rock the boat, so in past cases I simply bumped `-devel` but as an user it's a **pain** to be switching between `-devel` and non`-devel`. We don't want to go there.

However, it would be worth considering holding off bumping `mercurial` when there's a new feature release and waiting 1-2 weeks, to let the extensions catch up because they frequently break internal APIs. `hg-git` in particular breaks on virtually every one of these feature releases, and usually it takes a few ways for a fix to materialize upstream.

Reply:
I removed ```-devel``` from
- mercurial
- py-hgevolve
- py-hgsubversion

@lbschenkel, when this PR is close you will take the ownership of ```py-hggit``` and maintain the ```-devel``` as stated.

Reply:
> @lbschenkel, when this PR is close you will take the ownership of `py-hggit` and maintain the `-devel` as stated.

@mojca and I agree that there should be not `py-hggit-devel` port. If `py-hggit` is broken upstream but there's an unreleased fix, the fix should be backported or the port version bumped.

Reply:
I just noticed that there's a new maintainer for `py-hggit`, so I'm pinging @easye: please read the conversation above in this thread.

Reply:
Ops. I lost some lines ;-) sorry. Now I fix

Reply:
remove ```-devel``` from ```py-hggit```

Reply:
Just as I have mentioned, `py-hggit` is broken for Mercurial 4.8+. The latest release is from 2018-10-15 and Mercurial 4.8 has been released on 2018-11-02, which means the extension is broken for 4 months now and they haven't released a new version.

I have submitted a corrective PR here: https://github.com/macports/macports-ports/pull/3786